components/places/src/ffi.rs

```diff
@@ -147,6 +148,12 @@ impl PlacesConnection {
         Ok(f(&conn)?)
     }
 
+    fn new_interrupt_handle(&self) -> Result<Arc<SqlInterruptHandle>> {
```

Comment:
The semantics here also smell wrong. Given the arcs etc, I think the right approach would be for this to be called something like `get_interrupt_handle()` and the struct itself gets a `Weak<SqlInterruptHandle>` or something (the idea being there's exactly 1 interrupt handle per connection and getting multiple "new" ones per connection makes no sense) - but it doesn't make sense to fix this now, so please just add a comment referencing 1684 too.

(And if you get bored, starting to tackle 1684 makes some sense to me - we might decide to land it after we land this effort, but there's no good reason to not get started now.)

Reply:
I completely agree it feels weird to get a new handle everytime and will add proper comments,